flojoy/flojoy_cloud.py

```diff
@@ -0,0 +1,413 @@
+        ), '"c" dataset is not a float or int'
+        return dc
+
+
+class VectorModel(DefaultModel[DataC], Generic[DataC]):
+    data: Optional[DataC]
+
+    @validator("dataContainer")
+    def type_must_match(cls, dc):
+        assert dc["type"] == "Vector", "dataContainer type does not match."
+        return dc
+
+    @validator("dataContainer")
+    def keys_must_match(cls, dc):
+        assert "v" in dc, 'dataContainer does not contain "v" dataset.'
+        assert isinstance(dc["v"], list)
+        return dc
+
+
+class ImageModel(DefaultModel[DataC], Generic[DataC]):
+    data: Optional[DataC]
+
+    @validator("dataContainer")
+    def type_must_match(cls, dc):
+        assert dc["type"] == "Image", "dataContainer type does not match."
+        return dc
+
+    @validator("dataContainer")
+    def keys_must_match(cls, dc):
+        assert "r" in dc, 'dataContainer does not contain "r" dataset.'
+        assert "g" in dc, 'dataContainer does not contain "g" dataset.'
+        assert "b" in dc, 'dataContainer does not contain "b" dataset.'
+        assert isinstance(dc["r"], list), '"r" dataset is not a list'
+        assert isinstance(dc["g"], list), '"g" dataset is not a list'
+        assert isinstance(dc["b"], list), '"b" dataset is not a list'
+        return dc
+
+
+def check_deserialize(response):
+    dc_type = response["dataContainer"]["type"]
+    match dc_type:
+        case "OrderedPair":
+            OrderedPairModel.parse_obj(response)
+        case "OrderedTriple":
+            OrderedTripleModel.parse_obj(response)
+        case "DataFrame":
+            DataFrameModel.parse_obj(response)
+        case "Matrix":
+            MatrixModel.parse_obj(response)
+        case "Grayscale":
+            GrayscaleModel.parse_obj(response)
+        case "Scalar":
+            ScalarModel.parse_obj(response)
+        case "Vector":
+            ScalarModel.parse_obj(response)
+        case "Image":
+            ImageModel.parse_obj(response)
+        case _:
+            raise TypeError(
+                f"Unsupported DataContainer type: {dc_type}. Check case (e.g. OrderedPair)."
+            )
+
+
+class FlojoyCloud:
+    """
+    A class that allows pulling and pushing DataContainers from the
+    Flojoy cloud client (cloud.flojoy.ai).
+
+    Returns data in a pythonic format (e.g. Pillow for images,
+    DataFrames for arrays/matrices).
+
+    Will support the majority of the Flojoy cloud API:
+    https://rest.flojoy.ai/api-reference
+
+    Recommended for api key:
+    utils.get_credentials()[0]["value"]
+    or
+    os.environ.get("FLOJOY_CLOUD_KEY")
+    """
+
+    def __init__(self, apikey):
+        self.headers = {"api_key": apikey}
+        self.valid_types = [
+            "OrderedPair",
+            "OrderedTriple",
+            "DataFrame",
+            "Grayscale",
+            "Matrix",
+            "Scalar",
+            "Vector",
+            "Image",
+        ]
+
+    def create_payload(self, data, dc_type):
```

Comment:
This should be a "private" method, so just prefix it with "_" like "_create_payload"